components/server/ee/src/user/stripe-service.ts

```diff
@@ -130,6 +130,18 @@ export class StripeService {
         return session.url;
     }
 
+    async getPortalUrlForUser(user: User): Promise<string> {
+        const customer = await this.findCustomerByUserId(user.id);
+        if (!customer) {
+            throw new Error(`No Stripe Customer ID found for user '${user.id}'`);
+        }
+        const session = await this.getStripe().billingPortal.sessions.create({
+            customer: customer.id,
+            return_url: this.config.hostUrl.with(() => ({ pathname: `/t/${user.id}/billing` })).toString(),
```

Comment:
This `return_url` won't work. Isn't it simply `/billing`?

```suggestion
            return_url: this.config.hostUrl.with(() => ({ pathname: "/billing" })).toString(),
```